irs-api/src/test/java/org/eclipse/tractusx/irs/IrsWireMockIntegrationTest.java

```diff
@@ -371,13 +397,58 @@ void shouldCreateDetailedTombstoneForEdcErrors() {
         final Jobs jobForJobId = irsService.getJobForJobId(jobHandle.getId(), false);
 
         assertThat(jobForJobId.getJob().getState()).isEqualTo(JobState.COMPLETED);
+
         assertThat(jobForJobId.getShells()).isEmpty();
         assertThat(jobForJobId.getRelationships()).isEmpty();
         assertThat(jobForJobId.getSubmodels()).isEmpty();
-        assertThat(jobForJobId.getTombstones()).hasSize(1);
-        final Tombstone actualTombstone = jobForJobId.getTombstones().get(0);
-        assertThat(actualTombstone.getProcessingError().getRootCauses()).hasSize(1);
-        assertThat(actualTombstone.getProcessingError().getRootCauses().get(0)).contains("502 Bad Gateway");
+
+        final List<Tombstone> tombstones = jobForJobId.getTombstones();
+        assertThat(tombstones).hasSize(1);
+
+        final Tombstone actualTombstone = tombstones.get(0);
+        assertThat(actualTombstone.getBusinessPartnerNumber()).isEqualTo(TEST_BPN);
+        assertThat(actualTombstone.getEndpointURL()).isEqualTo(CONTROLPLANE_PUBLIC_URL);
+
+        final List<String> rootCauses = actualTombstone.getProcessingError().getRootCauses();
+        assertThat(rootCauses).hasSize(1);
+        assertThat(rootCauses.get(0)).contains("502 Bad Gateway");
+    }
+
+    @Test
+    void whenEmptyCatalogIsReturnedFromAllEndpoints() {
+        // Arrange
+        final String globalAssetId = "urn:uuid:334cce52-1f52-4bc9-9dd1-410bbe497bbc";
+        final List<String> edcUrls = List.of("https://test.edc1.io", "https://test.edc2.io");
+
+        WiremockSupport.successfulSemanticModelRequest();
+        WiremockSupport.successfulSemanticHubRequests();
+        WiremockSupport.successfulDiscovery(edcUrls);
+
+        edcUrls.forEach(edcUrl -> emptyCatalog(TEST_BPN, edcUrl));
+
+        // Act
+        final RegisterJob request = WiremockSupport.jobRequest(globalAssetId, TEST_BPN, 4);
+        final JobHandle jobHandle = irsService.registerItemJob(request);
+        assertThat(jobHandle.getId()).isNotNull();
+        waitForCompletion(jobHandle);
+        final Jobs jobForJobId = irsService.getJobForJobId(jobHandle.getId(), false);
+
+        // Assert
+
+        assertThat(jobForJobId.getJob().getState()).isEqualTo(JobState.COMPLETED);
+
+        assertThat(jobForJobId.getShells()).isEmpty();
+        assertThat(jobForJobId.getRelationships()).isEmpty();
+        assertThat(jobForJobId.getSubmodels()).isEmpty();
+
+        final List<Tombstone> tombstones = jobForJobId.getTombstones();
+        assertThat(tombstones).hasSize(1);
+
+        final Tombstone actualTombstone = tombstones.get(0);
+        assertThat(actualTombstone.getBusinessPartnerNumber()).isEqualTo(TEST_BPN);
+        assertThat(actualTombstone.getEndpointURL()).describedAs("Tombstone should contain all EDC URLs")
+                                                    .isEqualTo(String.join("; ", edcUrls));
+
```

Comment:
@ds-jhartmann when I add the following assertions
```         final List<String> rootCauses = actualTombstone.getProcessingError().getRootCauses();
        assertThat(rootCauses).hasSize(edcUrls.size());
        edcUrls.forEach(edcUrl -> assertThat(rootCauses).anyMatch(rootCause -> rootCause.contains(edcUrl)));
```
this fails. It seems that getting the root causes is somehow incomplete?!